### DISCUSSION TITLE
How to set foregroundcolor of a DataGrid cell in a DataGridTemplateColumn when the row is selected

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#12669] [Q&A] [answered]
With a regular DataGridTextColumn I could use a CellStyleClass to change the foreground color of a cell in a selected row:

          <DataGrid>					
            <DataGrid.Styles>
              
              <Style Selector="DataGridRow:selected DataGridCell.Current">
                <Setter Property="Foreground" Value="{Binding CurrentCellForeground}"/>
              </Style>
            </DataGrid.Styles>

            <DataGrid.Columns>
              <DataGridTextColumn Header="Name" Binding="{Binding Name}" Width="Auto"/>
              <DataGridTextColumn Header="Current" Binding="{Binding CurrentValue}" CellStyleClasses="Current"/>
            </DataGrid.Columns>						
          </DataGrid>

How can I achieve the same with a Label (or any) control in a DataGridTemplateColumn?

          <DataGrid>
            <DataGrid.Styles>
              
              <Style Selector="DataGridRow:selected DataGridCell.Current">
                <Setter Property="Foreground" Value="{Binding CurrentCellForeground}"/>
              </Style>
            </DataGrid.Styles>

            <DataGrid.Columns>
              <DataGridTextColumn Header="Name" Binding="{Binding Name}" Width="Auto"/>
              <DataGridTemplateColumn Header="Current" >
                <DataGridTemplateColumn.CellTemplate >
                  <DataTemplate >
                    <Label Content="{Binding CurrentValue}" /> 
                  </DataTemplate>
                </DataGridTemplateColumn.CellTemplate>
              </DataGridTemplateColumn>
            </DataGrid.Columns>
          </DataGrid>

**Comment by wgraaf** [comment, marked as answer]
Apparently, this was easy: the cell style class can be applied to the DataGridTemplateColumn:

              <DataGridTemplateColumn Header="Current" CellStyleClasses="Current">
                <DataGridTemplateColumn.CellTemplate >
                  <DataTemplate >
                    <Label Content="{Binding CurrentValue}" />
                  </DataTemplate>
                </DataGridTemplateColumn.CellTemplate>
                <DataGridTemplateColumn.CellEditingTemplate>
                  <DataTemplate >
                    <TextBox Text="{Binding CurrentValue, Mode=OneTime}" Background="Yellow" CaretIndex="9999" />
                  </DataTemplate>
                </DataGridTemplateColumn.CellEditingTemplate>
              </DataGridTemplateColumn>

**Comment by timunie** [comment]
You can also do 

```xml
<Style Selector="DataGridCell.MyClass TextBlock"> 
... 
</Style>
```

to style any child control matching your class :-)
